### PR TITLE
Check and update version for transactional client

### PR DESCRIPTION
**Pull request by KazankovMarch**
This is a common usage of TransactionalClient, when we want to sync separated servers by zookeeper. Let's make a method with a readable name

### CODE REVIEW

File: jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java [hunk 1]
```diff
@@ -95,4 +105,40 @@ public void testMixedCreateDelete_Failure() throws Exception {
...
+            final int numberOfTransaction = i;
+            futures.add(CompletableFuture.runAsync(() -> {
+                try {
+                    TransactionalClient.tryCommit(zkTestingServer.getClient(), 1, transaction -> {
+                        transaction.checkAndUpdateVersion(lockPath);
+                        transaction.createPath("/" + numberOfTransaction);
+                        countDownLatch.countDown();
+                        countDownLatch.await();
+                    });
+                    expectedNodes.add(String.valueOf(numberOfTransaction));
```
Comment (swarmshine) [inline jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java]: expectedNodes is not threadsafe

File: jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java [hunk 2]
```diff
@@ -95,4 +105,40 @@ public void testMixedCreateDelete_Failure() throws Exception {
...
+        Set<String> expectedNodes = new HashSet<>();
+        expectedNodes.add("lock");
+
+        int quantityOfTransactions = 10;
+        CountDownLatch countDownLatch = new CountDownLatch(quantityOfTransactions);
+        List<CompletableFuture<Void>> futures = new ArrayList<>(quantityOfTransactions);
+        ExecutorService executor = Executors.newFixedThreadPool(quantityOfTransactions);
+
+        for (int i = 0; i < quantityOfTransactions; i++) {
+            final int numberOfTransaction = i;
```
Comment (swarmshine) [inline jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java]: transactionNumber

File: jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java [hunk 3]
```diff
@@ -78,6 +78,21 @@ public TransactionalClient createPathWithParentsIfNeeded(String path) throws Exc
         return this;
     }
 
+    /**
+     * Usage: invoke this method with the same parameter during different transactions,
+     * which must not succeed in parallel.
+     *
```
Comment (swarmshine) [inline jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java:85]: better to add notes to javadoc: 

node by given path should exist, value of given node will be overwritten

**Review by swarmshine — CHANGES_REQUESTED** [review]

File: jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java (hunk 2 above)
Comment (KazankovMarch) [reply jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java]: fixed

File: jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java (hunk 3 above)
Comment (KazankovMarch) [reply jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java:85]: added

File: jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java (hunk 2 above)
Comment (KazankovMarch) [reply jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java]: fixed

File: jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java (hunk 1 above)
Comment (KazankovMarch) [reply jfix-zookeeper/src/test/java/ru/fix/zookeeper/transactional/TransactionalClientIT.java]: HashSet replaced with CopyOnWriteArraySet

File: jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java (hunk 3 above)
Comment (surkovich) [reply jfix-zookeeper/src/main/java/ru/fix/zookeeper/transactional/TransactionalClient.java:85]: Actually, this javadoc doesn't explain what the method is doing. 
"invoke this method with the same parameter during different transactions" - does it mean, it should always be used with the same parameters? 
As I see, it should be something like "Adds an updating version operation at the given path. Update version - operation, that <...> at the given path <...>. Path should exists, <what happens> elsewhere."

**Pull request merged**